src/reflect/scala/reflect/io/ZipArchive.scala

```diff
@@ -62,6 +62,25 @@ object ZipArchive {
       if (front) path.substring(0, idx + 1)
       else path.substring(idx + 1)
   }
+  def pathToDotted(path: String): String = {
+    if (path == "/") ""
+    else {
+      val slashEnd = path.endsWith("/")
+      val len = path.length - (if (slashEnd) 1 else 0)
+      val result = new Array[Char](len)
+      var i = 0
+      while (i < len) {
+        val char = path.charAt(i)
+        result(i) = if (char == '/') '.' else char
+        i += 1
+      }
+      new String(result)
+    }
+  }
+  def dottedToPath(dotted: String): String = {
```

Comment:
make this a local method in the rhs of `lazy val allDirs` to avoid accidental uses?